server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -221,15 +227,15 @@ public RemoteClusterStateManifestInfo writeFullMetadata(ClusterState clusterStat
             clusterState,
             new ArrayList<>(clusterState.metadata().indices().values()),
             emptyMap(),
-            clusterState.metadata().customs(),
+            RemoteGlobalMetadataManager.filterCustoms(clusterState.metadata().customs(), isPublicationEnabled),
```

Comment:
isPublicationENabled - can we pass it in constructor to Global Manager and then it can decide which ones to upload?

Reply:
We discussed this earlier as well that these entity managers have to be refactored with a defined interface.
RemoteClusterStateService will maintain registry of all the managers and then call these managers for upload and download. RemoteClusterStateService should only be responsible for orchestration of read and write of the state and never deal with individual entities. Individual entities should be owned by their respective managers.

Reply:
I can take this refactoring as a follow up.